MhLabs.AwsSignedHttpClient/AwsSignedHttpMessageHandlerWithLogging.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using MhLabs.AwsSignedHttpClient.Credentials;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace MhLabs.AwsSignedHttpClient
+{
+    public class AwsSignedHttpMessageHandlerWithLogging<TClient> : AwsSignedHttpMessageHandler
+    {
+        private readonly ILogger<TClient> _logger;
+
+        public AwsSignedHttpMessageHandlerWithLogging(ILogger<TClient> logger, ICredentialsProvider credentialsProvider) : base(credentialsProvider)
+        {
+            _logger = logger ?? NullLogger<TClient>.Instance;
+        }
+
+        public AwsSignedHttpMessageHandlerWithLogging(ILogger<TClient> logger)
```

Comment:
Naming methods/classes efter parameters or behavior is not good practice

Reply:
- Agree. If you can come up with something better please change it. I first tried the decorator pattern but got complex pretty quick.
- It will get resolved automatically via IoC.
- If the ILoggerFactory is null, then there is none registered in IoC and that's why it should not really log anything, just as it won't for other classes and libs using ILoggerFactory/ILogger.

Reply:
Like I've said I think we should implement a default version for ILogger if none is configured. If we do that we wouldn't need two versions of the message handler, just one that performs logging regardless of ILogger implementation?